### PR TITLE
[Bugfix][Relay][Keras] Fix the wrong implementation logic about cropping2D

### PR DESCRIPTION
**Pull request by jikechao**
This PR fixed the wrong calculation logic which will lead to wrong inference results.
The original cropping algorithm shown below is incorrect.
image
I modified the code to fix it. Meanwhile, a test case that can trigger this bug was added to the unit test. 

### Step to reproduce
```
import tvm
import tvm.relay as relay
import numpy as np
from tensorflow import keras
from tensorflow.keras import layers, models
input_shape = (2, 9, 9, 2)
input_data = np.random.random(size=input_shape)
x = layers.Input(shape=input_shape[1:], dtype='float32')

layer = keras.layers.Cropping2D(cropping=[2,1], data_format='channels_last')
layer.set_weights(layer.get_weights())

y = layer(x)
model = models.Model(x, y)
model.summary()
res_keras = model(input_data)

shape_dict = {'input_1': input_shape}
mod, params = relay.frontend.from_keras(model, shape_dict)
with tvm.transform.PassContext(opt_level=3):
    model = relay.build_module.create_executor("graph", mod, tvm.cpu(0), 'llvm', params).evaluate()

test_x_tvm = input_data
res_tvm = model(tvm.nd.array(test_x_tvm.astype('float32'))).numpy()

np.testing.assert_allclose(res_keras, res_tvm, atol=1e-3, rtol=1e-3)

```

image

cc @echuraev @masahi @Hzfengsy

### CODE REVIEW

**Review by echuraev — APPROVED** [review]

**Comment by jikechao** [comment]
2 existing test cases fail after this fix in the above CI approach. However, these 2 test cases can run successfully with this patch. 
@echuraev, Could you help me check why these test cases failed?


The Stack trace is here:
```
Error
AssertionError: 
Not equal to tolerance rtol=1e-07, atol=0
Mismatched elements: 3 / 4 (75%)
Max absolute difference: 7
Max relative difference: 0.21875
 x: array([ 1, 23, 25,  3])
 y: array([ 1, 25, 32,  0])
Stacktrace
self = <test_forward.TestKeras object at 0xffff65871640>
keras_mod = <module 'keras.api._v2.keras' from '/venv/apache-tvm-py3.8/lib/python3.8/site-packages/keras/api/_v2/keras/__init__.py'>
    def test_forward_crop(self, keras_mod):
        """test_forward_crop"""
        data = keras_mod.layers.Input(shape=(32, 32, 3))
        x = keras_mod.layers.Cropping2D(cropping=((1, 1), (1, 1)))(data)
        x = keras_mod.layers.Cropping2D(cropping=(1, 1))(x)
        x = keras_mod.layers.Cropping2D(cropping=1)(x)
        x = keras_mod.layers.Cropping2D(cropping=((0, 1), (1, 0)))(x)
        x = keras_mod.layers.Cropping2D(cropping=(1, 0))(x)
        x = keras_mod.layers.Cropping2D(cropping=0)(x)
        x = keras_mod.layers.Add()([x, x])
        keras_model = keras_mod.models.Model(data, x)
>       verify_keras_frontend(keras_model)
tests/python/frontend/keras/test_forward.py:444: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
tests/python/frontend/keras/test_forward.py:118: in verify_keras_frontend
    tvm.testing.assert_allclose(kout, tout, rtol=1e-5, atol=1e-5)
```

**Comment by masahi** [comment]
See https://github.com/apache/tvm/pull/14983#issuecomment-1583221170 for the arm build issue.

**Comment by masahi** [comment]
@tvm-bot rerun

**Comment by jikechao** [comment]
The initial patch will lead to existing test cases crashing. So, I updated the patch to fix it.

File: tests/python/frontend/keras/test_forward.py [hunk 1]
```diff
@@ -443,6 +443,11 @@ def test_forward_crop(self, keras_mod):
         keras_model = keras_mod.models.Model(data, x)
         verify_keras_frontend(keras_model)
 
+        x = keras_mod.layers.Cropping2D(cropping=(2, 1))(data)
+        x = keras_mod.layers.Cropping2D(cropping=(1, 2))(x)
+        keras_model = keras_mod.models.Model(data, x)
+        verify_keras_frontend(keras_model, layout="NHWC")
```
Comment (echuraev) [inline tests/python/frontend/keras/test_forward.py:450]: Could you also please add a test for `NCHW` layout?

Comment (jikechao) [reply tests/python/frontend/keras/test_forward.py:451]: Thanks for your reminder, related test cases have been added.

**Comment by jikechao** [comment]
@tvm-bot rerun

**Comment by jikechao** [comment]
@tvm-bot rerun

**Comment by jikechao** [comment]
@tvm-bot rerun

**Comment by jikechao** [comment]
@tvm-bot rerun

**Comment by jikechao** [comment]
@tvm-bot rerun

**Comment by jikechao** [comment]
@tvm-bot rerun

**Comment by jikechao** [comment]
@echuraev @masahi CI test has finished. Could you help me to merge this PR? Thanks!

**Review by echuraev — APPROVED** [review]

**Pull request merged**